lib/spack/spack/util/ld_so_conf.py

```diff
@@ -121,18 +123,18 @@ def host_dynamic_linker_search_paths():
         # to its prefix.
         if elf.has_pt_interp:
             dynamic_linker = elf.pt_interp_str.decode("utf-8")
-            dynamic_linker_name = os.path.basename(dynamic_linker)
+            dynamic_linker_name = PurePath(dynamic_linker).name
```

Comment:
This instance and below might want to be `Path` objects instead

Reply:
This got switched to `Path`. The construction is in a different spot though after refactoring from another suggestion.